prometheus-to-sd/config/dynamic_source.go

```diff
@@ -71,33 +87,81 @@ func createOptionsForPodSelection(nodeName string, sources map[string]url.URL) v
 	for key := range sources {
 		nameList += key + ","
 	}
-	labelSelector := fmt.Sprintf("%s in (%s)", nameLabel, nameList)
-	return v1.ListOptions{
+	labelSelector := fmt.Sprintf("%s in (%s)", componentNameLabel, nameList)
+	listOptions := v1.ListOptions{
 		FieldSelector: fmt.Sprintf("spec.nodeName=%s", nodeName),
 		LabelSelector: labelSelector,
 	}
+	glog.Infof("Filters for dynamic components: %+v", listOptions)
+	return listOptions
 }
 
-func getConfigsFromPods(pods []core.Pod, sources map[string]url.URL) []SourceConfig {
-	var sourceConfigs []SourceConfig
-	for _, pod := range pods {
-		componentName := pod.Labels[nameLabel]
-		source, _ := sources[componentName]
-		podConfig := PodConfig{
-			PodId:       pod.Name,
-			NamespaceId: pod.Namespace,
-		}
-		sourceConfig, err := mapToSourceConfig(componentName, source, pod.Status.PodIP, podConfig)
-		if err != nil {
-			glog.Warning("could not create source config for pod %s: %v", pod.Name, err)
-		}
-		sourceConfigs = append(sourceConfigs, *sourceConfig)
+func getConfigFromPod(pod core.Pod, sources map[string]url.URL) (*SourceConfig, error) {
+	if pod.Status.PodIP == "" {
+		return nil, errors.New("pod ip is missing")
```

Comment:
This will be triggered for every new pod that is waiting to be scheduled. Maybe we should silence it ?